module/zfs/vdev.c

```diff
@@ -1672,6 +1672,35 @@ vdev_set_deflate_ratio(vdev_t *vd)
 	}
 }
 
+/*
+ * Maximize performance by inflating the configured ashift for top level
+ * vdevs to be as close to the physical ashift as possible while maintaining
+ * administrator defined limits and ensuring it doesn't go below the
+ * logical ashift.
+ */
+static void
+vdev_ashift_optimize(vdev_t *vd)
+{
+	ASSERT(vd == vd->vdev_top);
+
+	if (vd->vdev_ashift < vd->vdev_physical_ashift) {
+		vd->vdev_ashift = MIN(
+		    MAX(zfs_vdev_max_auto_ashift, vd->vdev_ashift),
+		    MAX(zfs_vdev_min_auto_ashift,
+		    vd->vdev_physical_ashift));
+	} else {
+		/*
+		 * Unusual case where logical ashift > physical ashift
+		 * so we can't cap the calculated ashift based on max
```

Comment:
Reading the code, this comment looks wrong: what about the case where logical = physical?

Reply:
Agree that the comment is confusing since it only talks about the unusual case but doesn't mention the most common case. Since I copied this from the original commit, I suspect the author was trying to highlight just the unusual case. I will rework the comment. Thanks for the review!